File: design-patterns/go/builder/pizza/types/waiter.go
package types

type Waiter struct {
	pizzaBuilder PizzaBuilder
}

func NewWaiter(pz PizzaBuilder) *Waiter {
	return &Waiter{
		pizzaBuilder: pz,
	}
}

func (w *Waiter) GetPizza() Pizza {
	return w.pizzaBuilder.GetPizza()
}

func (w *Waiter) SetPizzaBuilder(pb PizzaBuilder) {
	w.pizzaBuilder = pb
}

func (w *Waiter) CookPizza() {
	w.pizzaBuilder.UseDough()
	w.pizzaBuilder.UseSauce()
	w.pizzaBuilder.AddToppings()
}

File: design-patterns/java/builder/Waiter.java
class Waiter {
    private PizzaBuilder pizzaBuilder;

    public void setPizzaBuilder(PizzaBuilder pb) { pizzaBuilder = pb; }

    public Pizza getPizza() { return pizzaBuilder.getPizza(); }

    public void constructPizza() {
        pizzaBuilder.buildDough();
        pizzaBuilder.buildSauce();
        pizzaBuilder.buildTopping();
    }
}